src/action/list_directory.rs

```diff
@@ -0,0 +1,689 @@
+
+    type Proto = ListDirRequest;
+
+    fn from_proto(proto: Self::Proto) -> Result<Request, session::ParseError> {
+        let missing = session::MissingFieldError::new;
+        let pathspec = proto.pathspec.ok_or(missing("path spec"))?;
+        let path_type = pathspec.pathtype
+            .ok_or(missing("path type"))?;
+        if path_type != PathType::Os as i32 {
+            return Err(unsupported_value(String::from("path type"),
+                                         path_type.to_string()));
+        }
+        let path_option = pathspec.path_options
+            .unwrap_or(Options::CaseLiteral as i32);
+        if path_option != Options::CaseLiteral as i32 {
+            return Err(unsupported_value(String::from("path option"),
+                                         path_option.to_string()));
+        };
+        Ok(Request {
+            path: get_path(&pathspec.path),
+        })
+    }
+}
+
+fn get_time_since_unix_epoch(sys_time: &Option<SystemTime>) -> Option<u64> {
+    return sys_time.map_or(None, |time| time.duration_since(UNIX_EPOCH)
+        .map_or(None, |dur| Some(dur.as_secs())));
+}
+
+impl super::Response for Response {
+
+    const RDF_NAME: Option<&'static str> = Some("StatEntry");
+
+    type Proto = StatEntry;
+
+    fn into_proto(self) -> Self::Proto {
+        StatEntry {
+            st_mode: self.mode,
+            st_ino: self.ino,
+            st_dev: self.dev,
+            st_nlink: self.nlink,
+            st_uid: self.uid,
+            st_gid: self.gid,
+            st_size: self.size,
+            st_atime: get_time_since_unix_epoch(&self.atime),
+            st_mtime: get_time_since_unix_epoch(&self.mtime),
+            st_ctime: get_time_since_unix_epoch(&self.ctime),
+            st_blocks: self.blocks,
+            st_blksize: self.blksize,
+            st_rdev: self.rdev,
+            st_flags_osx: None,
+            st_flags_linux: self.flags_linux,
+            symlink:
+            self.symlink
+                .map(|symlink| symlink.to_string_lossy().to_string()),
+            registry_type: None,
+            resident: None,
+            pathspec: Some(rrg_proto::PathSpec {
+                // Represents `CaseLiteral` path option (other options are not
+                // supported).
+                path_options: Some(Options::CaseLiteral as i32),
+                // Represents `OS` path type (other types are not supported).
+                pathtype: Some(PathType::Os as i32),
+                path: Some(self.path.to_string_lossy().to_string()),
+                ..Default::default()
+            }),
+            registry_data: None,
+            st_crtime: get_time_since_unix_epoch(&self.crtime),
+            ext_attrs: vec![],
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::action::Request;
+    use tempfile::tempdir;
+
+    #[cfg(target_os = "linux")]
+    use std::os::unix::fs::MetadataExt;
+
+    /// Fills ListDirRequest with provided fields
```

Comment:
`ListDirRequest` should be backticked. Also, a missing dot at the end.

Reply:
Fixed.